### PR TITLE
Use MFEM_INSTALL_DIR to find occa kernel locations [okina-occa-paths]

### PR DESCRIPTION
**Pull request by dmed256**
- Removed `MFEM_REAL_DIR` and replaced it with `MFEM_DIR`

- Adds `MFEM_DIR` to the `config.hpp` file to find OCCA kernel locations

- OCCA kernels use the `occa://mfem` prefix for mfem kernels

  - `fem/occa.okl` &rarr; `occa://mfem/fem/occa.okl`

### CODE REVIEW

**Comment by dmed256** [comment]
@v-dobrev @tzanio Any objections to the `MFEM_DIR` update?

**Comment by v-dobrev** [comment]
I object, this is not how the mfem directory should be saved in the built library.

**Comment by dmed256** [comment]
@v-dobrev In that case, any suggestions?

**Comment by v-dobrev** [comment]
Similar need was handled in the engines-dev branch. You need to save the mfem source directory (for builds before install) and the mfem install directory.

Another issue: in `config.mk`, `MFEM_DIR` is expected to be set outside, before including it. You should use another name. e.g. `MFEM_SOURCE_DIR` and `MFEM_INSTALL_DIR`.

**Comment by v-dobrev** [comment]
See e.g. https://github.com/mfem/mfem/blob/engines-dev/makefile, https://github.com/mfem/mfem/blob/engines-dev/config/config.mk.in, https://github.com/mfem/mfem/blob/engines-dev/config/config.hpp.in - search for `MFEM_SOURCE_DIR` and `MFEM_INSTALL_DIR`.

File: config/defaults.mk [hunk 1]
```diff
@@ -27,7 +27,7 @@ DEBUG_FLAGS = -g $(MFEM_XCOMPILER) -Wall $(BASE_FLAGS)
 
 # Destination location of make install
 # PREFIX = $(HOME)/mfem
-PREFIX = ./mfem
+PREFIX = .
```
Comment (dmed256) [inline config/defaults.mk]: @v-dobrev Changed this to the current directory since that's the default installation path

Comment (dmed256) [reply config/defaults.mk]: Nevermind, @tzanio mentioned why it's `./mfem`. Reverting this change

File: makefile [hunk 2]
```diff
@@ -98,7 +98,8 @@ $(if $(wildcard $(THIS_MK)),,$(error Makefile not found "$(THIS_MK)"))
 MFEM_DIR ?= $(patsubst %/,%,$(dir $(THIS_MK)))
 MFEM_REAL_DIR := $(realpath $(MFEM_DIR))
 $(if $(MFEM_REAL_DIR),,$(error Source directory "$(MFEM_DIR)" is not valid))
-SRC := $(if $(MFEM_REAL_DIR:$(CURDIR)=),$(MFEM_DIR)/,)
+override MFEM_DIR := $(MFEM_REAL_DIR)
+SRC := $(if $(MFEM_DIR:$(CURDIR)=),$(MFEM_DIR)/,)
```
Comment (v-dobrev) [inline makefile]: I want to keep `MFEM_DIR` and `MFEM_REAL_DIR` as they are - this change is not really needed here. Or is it?

Comment (camierjs) [reply makefile]: The last commit should revert the `MFEM_DIR` changes

**Pull request merged**